servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Completable.java

```diff
@@ -757,8 +798,45 @@ public final Completable beforeCancel(Runnable onCancel) {
      * </ul>
      * for Subscriptions/{@link Subscriber}s of the returned {@link Completable}. <strong>MUST NOT</strong> throw.
      * @return The new {@link Completable}.
+     *
+     * @see <a href="http://reactivex.io/documentation/operators/do.html">ReactiveX do operator.</a>
      */
     public final Completable beforeFinally(Runnable doFinally) {
+        return beforeFinally(TerminalSignalConsumer.from(doFinally));
+    }
+
+    /**
+     * Invokes the {@code whenFinally} {@link Runnable} argument <strong>before</strong> any of the following terminal
+     * methods are called:
+     * <ul>
+     *     <li>{@link Subscriber#onComplete()} - invokes {@link TerminalSignalConsumer#onComplete()}</li>
+     *     <li>{@link Subscriber#onError(Throwable)} - invokes {@link TerminalSignalConsumer#onError(Throwable)}</li>
+     *     <li>{@link Cancellable#cancel()} - invokes {@link TerminalSignalConsumer#onCancel()}</li>
+     * </ul>
+     * for Subscriptions/{@link Subscriber}s of the returned {@link Completable}.
+     * <p>
+     * From a sequential programming point of view this method is roughly equivalent to the following:
+     * <pre>{@code
+     *  try {
+     *      resultOfThisCompletable();
+     *  } finally {
+     *      doFinally.onTerminalSignal(); // Invokes an appropriate terminal signal
```

Comment:
Similar comment for other methods here and in other sources.

The semantics here do not really match `try-finally` now as the code paths are different for error vs completion:

```java
     *  try {
     *      resultOfThisCompletable();
     *  } catch(Throwable t) {
     *      doFinally.onError(t);
     *      nextOperation(); // Maybe notifying of cancellation, or termination
     *      return;
     *  }
     *  doFinally.onComplete(); 
     *  nextOperation(); // Maybe notifying of cancellation, or termination
```